text/evaluation/smollm3/README.md

```diff
@@ -0,0 +1,79 @@
+# SmolLM3-3B evaluation scripts
+
+We use the [LightEval](https://github.com/huggingface/lighteval/) library to benchmark our models.
+
+## Setup
+
+Use conda/uv/venv with `python>=3.11`.
+
+For reproducibility, we recommend fixed versions of the libraries:
+
+```sh
+pip install uv
+uv venv smol3_venv --python 3.11 
+source smol3_venv/bin/activate
+
+GIT_LFS_SKIP_SMUDGE=1 uv pip install -r requirements.txt
+```
+
+## Running the evaluations
+
+All commands below were run on 2 x H100s with 80GB of memory each, using the `vllm` backend.
+
+### SmolLM3-3B base model
+
+```bash
+MODEL_ARGS="model_name=HuggingFaceTB/SmolLM3-3B-Base,dtype=bfloat16,max_model_length=32768,max_num_batched_tokens=32768,generation_parameters={temperature:0},tensor_parallel_size=2,gpu_memory_utilization=0.7"
+lighteval vllm \
+    "$MODEL_ARGS" \
+    "smollm3_base.txt" \
+    --custom-tasks "tasks.py" \
+    --output-dir "evals/" \
+    --save-details
+```
+
+### SmolLM3-3B mid-trained model
+
+To evaluate the mid-trained reasoning model, first install a dedicated branch of LightEval that supports the language families of SmolLM3:
+
+```sh
+uv pip uninstall lighteval
+GIT_LFS_SKIP_SMUDGE=1 uv pip install "lighteval[extended_tasks,multilingual] @ git+https://github.com/huggingface/lighteval.git@2788496c3ef9973a24c7d8b87ee5b2cceb430872"
```

Comment:
This commit is needed because the one in `requirements.txt` doesn't have some important fixes we added for IFEval & reasoning models. I will run a few base model evals with this commit and if they look OK we can just use this commit as the source of truth